### ISSUE TITLE
LDAP login regression

### ISSUE THREAD

**Issue by alex1989hu** [goharbor/harbor#13140]
**Expected behavior and actual behavior:**
The LDAP configuration works with `v2.0.2` and latter releases of `v1.x.x`. The login just takes few seconds; there is neither warning nor error messagein `core`. The `v2.1.0` produce lot warning and error messages, additionally the LDAP login takes ~20 seconds.

# Original
Works with `v2.0.2` and latter releases of `v1.x.x`.

`memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe`

**Result**: I can do log in, I have my permission but I see these in `core` logs:

```
2020-09-21T07:27:36Z [ERROR] [/common/utils/ldap/ldap.go:302]: Wrong filter format, filter:()                                                                                                                    │
2020-09-21T07:27:36Z [WARNING] [/core/auth/ldap/ldap.go:117]: Can not get the ldap group name with DN CN=<redacted-1>, error invalid filter syntax
2020-09-21T07:27:36Z [WARNING] [/core/auth/ldap/ldap.go:117]: Can not get the ldap group name with DN CN=<redacted-2>, error invalid filter syntax
...
```

## LDAP Filter v2

Lets modify the `LDAP Filter`, mod: `'....'`

`'memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe'`

**Result:** I can do log in, I have my permission but I see the same error and warming messages *(see original)* in `core` logs.

## LDAP Filter v3

Lets modify the `LDAP Filter`, mod: `'( .... )'`

`'(memberOf=CN=APP_REGISTRY,OU=Managed,OU=Groups,OU=Enterprise,dc=foo,dc=bar,dc=john,dc=doe)'`

**Result:**  I can not do a login, `core` logs:

```
2020-09-21T07:35:34Z [WARNING] [/core/auth/ldap/ldap.go:65]: ldap search fail: invalid filter syntax                                                                                                             │
2020-09-21T07:35:34Z [ERROR] [/core/controllers/base.go:109]: Error occurred in UserLogin: invalid filter syntax
```

=> **KO**

**Steps to reproduce the problem:**
Use `v2.1.0`

**Versions:**
Please specify the versions of following systems.

- harbor version: v2.1.0
- docker engine version: N/A
- containerd version: 1.3.3
- docker-compose version: N/A

**Additional context:**
Related PR: https://github.com/goharbor/harbor/pull/12997
- **Harbor config files:**
See above.
- **Log files:**
See above.

**Comment by alex1989hu** [comment]
**UPDATE:**
Sometimes not able to do a login, got a message on the UI: `Core service is not available.` + prints same error, warning messages in `core`.

**Comment by bitsf** [comment]
use `docker ps` to check if harbor-core is always restarted

**Comment by alex1989hu** [comment]
/cc @bitsf 
There is no `docker`, it is running in `Kubernetes` && `containerd`.
There was only one restart since the early start. 

See details below:

```bash
kubectl -n harbor describe pod harbor-harbor-core-65968c4676-8vzqd
Name:         harbor-harbor-core-65968c4676-8vzqd
Namespace:    harbor
Priority:     0
Node:         prod-k8s-worker-f08f772d/3.249.66.187
Start Time:   Mon, 21 Sep 2020 09:05:58 +0200
Labels:       app=harbor
              component=core
              pod-template-hash=65968c4676
              release=harbor
Annotations:  checksum/configmap: 566853c00e131367786098e2d25cc4a677f3368adcaf033e101fc65d12bc7f78
              checksum/secret: 2c13973026766c7dfde5e44af9e80fef753317c087191febc21d84f974b9ef86
              checksum/secret-jobservice: 22c03c7f359109ff5eb766d31ce31911c0ecf8969e81867221f98fe9d19c752b
              kubernetes.io/psp: privileged
Status:       Running
IP:           10.0.3.233
IPs:
  IP:           10.0.3.233
Controlled By:  ReplicaSet/harbor-harbor-core-65968c4676
Containers:
  core:
    Container ID:   containerd://290f7dd1ecda7c84a917763fdbb67c1229b08b031575a9b25d002b93e264eb6d
    Image:          goharbor/harbor-core:v2.1.0
    Image ID:       docker.io/goharbor/harbor-core@sha256:f641c5f10514f65b8b9cb4bf8c5936e78147987e62d3cf3c4712d057e1f9421f
    Port:           8080/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Mon, 21 Sep 2020 09:07:25 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Mon, 21 Sep 2020 09:06:07 +0200
      Finished:     Mon, 21 Sep 2020 09:07:22 +0200
    Ready:          True
    Restart Count:  1
    Limits:
      cpu:     1500m
      memory:  512Mi
    Requests:
      cpu:      100m
      memory:   256Mi
    Liveness:   http-get http://:8080/api/v2.0/ping delay=300s timeout=1s period=10s #success=1 #failure=3
    Readiness:  http-get http://:8080/api/v2.0/ping delay=20s timeout=1s period=10s #success=1 #failure=3
    Environment Variables from:
      harbor-harbor-core  ConfigMap  Optional: false
      harbor-harbor-core  Secret     Optional: false
    Environment:
      CORE_SECRET:        <set to the key 'secret' in secret 'harbor-harbor-core'>                   Optional: false
      JOBSERVICE_SECRET:  <set to the key 'JOBSERVICE_SECRET' in secret 'harbor-harbor-jobservice'>  Optional: false
    Mounts:
      /etc/core/app.conf from config (rw,path="app.conf")
      /etc/core/key from secret-key (rw,path="key")
      /etc/core/private_key.pem from token-service-private-key (rw,path="tls.key")
      /etc/core/token from psc (rw)
      /harbor_cust_cert/custom-ca.crt from ca-bundle-certs (rw,path="ca.crt")
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-dvzft (ro)
Conditions:
  Type              Status
  Initialized       True
  Ready             True
  ContainersReady   True
  PodScheduled      True
Volumes:
  config:
    Type:      ConfigMap (a volume populated by a ConfigMap)
    Name:      harbor-harbor-core
    Optional:  false
  secret-key:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-harbor-core
    Optional:    false
  token-service-private-key:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-harbor-core
    Optional:    false
  psc:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:
    SizeLimit:  <unset>
  ca-bundle-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  harbor-ge-ext-cert
    Optional:    false
  default-token-dvzft:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-dvzft
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:          <none>
```

**Comment by moshloop** [comment]
I believe this is due to https://github.com/goharbor/harbor/blob/master/src/core/auth/ldap/ldap.go#L114-L125 doing an LDAP lookup for each group a user is a member of, It should probably be checking the DB first and then only doing the LDAP lookup and/or using a lookup cache

**Comment by xaleeks** [comment]
@stonezdj please check this out

**Comment by stonezdj** [comment]
It is already fix by PR https://github.com/goharbor/harbor/pull/13298

**Comment by xaleeks** [comment]
awesome!!

**Comment by xaleeks** [comment]
is this in the 2.1.1? or the following patch? @stonezdj

**Comment by xaleeks** [comment]
this is fixed in the 2.1.1 coming next week @moshloop
